internal/provider/edgegw/nat_rule_schema.go

```diff
@@ -0,0 +1,266 @@
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+					// TODO - Validator of IP or IP/CIDR
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"app_port_profile_id": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "Application Port Profile ID to which the rule applies.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"app_port_profile_name": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "Application Port Profile Name to which the rule applies.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"dnat_external_port": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "This represents the external port number or port range when doing DNAT port forwarding from external to internal. If not specify, all ports are translated",
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+					Validators: []validator.String{
+						fstringvalidator.RequireIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("DNAT")}),
+						fstringvalidator.RequireIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("NO_DNAT")}),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"snat_destination_address": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The destination addresses to match in the SNAT Rule. This must be supplied as a single IP or Network CIDR. Providing no value for this field results in match with ANY destination network.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+					Validators: []validator.String{
+						fstringvalidator.RequireIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("SNAT")}),
+						fstringvalidator.RequireIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("NO_SNAT")}),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			// Not Supported on CloudAvenue
+			// "logging": superschema.SuperBoolAttribute{
+			// 	Common: &schemaR.BoolAttribute{
+			// 		MarkdownDescription: "Enable to have the address translation performed by this rule logged ",
+			// 	},
+			// 	Resource: &schemaR.BoolAttribute{
+			// 		Optional: true,
+			// 		Default:  booldefault.StaticBool(true),
+			// 	},
+			// 	DataSource: &schemaD.BoolAttribute{
+			// 		Computed: true,
+			// 	},
+			// },
```

Comment:
```suggestion
```

Reply:
I let this comment in order to explain why this attribute is not implemented